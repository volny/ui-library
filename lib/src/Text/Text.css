.Text {
  composes: sansSerif antialiased from "../typography.css";
  composes: fontWeightNormal from "../typography.css";
  line-height: 20px;
}

.paragraph {
  composes: Text;
  font-size: 16px;
}

.caption {
  composes: Text;
  font-size: 12.8px;
}
