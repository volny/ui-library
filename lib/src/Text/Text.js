import React from 'react'
import PropTypes from 'prop-types'
import styled, { css, withComponent } from 'styled-components'

import { fontStyle } from '../typography.js'
import colors from '../colors.js'

const Styled = styled`
  ${fontStyle}
  ${({ color, align, italic, bold, caption }) => `
    color: ${color ? colors[color] : colors.darkGrey};
    line-height: 20px;
    text-align: ${align};
    font-style: ${italic ? 'italic' : 'normal'};
    font-weight: ${bold ? 'bold' : '300'};
    font-size: ${caption ? '12.8px' : '16px'};
    letter-spacing: ${caption ? '.5px' : 'normal'};
  `}
`

const Text = props => {
  const Tag = Styled.withComponent(props.inline ? 'span' : 'p')
  return <Tag {...props}>{props.children}</Tag>
}

Text.propTypes = {
  children: PropTypes.node.isRequired,
  align: PropTypes.oneOf(['left', 'right', 'center', 'justify']),
  color: PropTypes.oneOf([
    'darkGrey',
    'white',
    'green',
    'darkGrey70',
    'darkGrey30',
    'darkGrey10',
    'white90',
    'white30',
    'green30',
    'green20',
    'yellow',
    'red',
  ]),
  inline: PropTypes.bool,
  italic: PropTypes.bool,
  bold: PropTypes.bool,
  caption: PropTypes.bool,
}

export default Text
