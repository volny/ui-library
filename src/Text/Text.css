.paragraph {
  font-size: 16px;
  font-weight: 300;
  line-height: 20px;
}

.caption {
  font-size: 12.8px;
  font-weight: 300;
  line-height: 12px;
}
