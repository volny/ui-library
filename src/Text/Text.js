// @flow

import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import styles from './Text.css';
import typography from '../typography.css';
import colors from '../colors.css';

const Text = ({
  children,
  align = 'left',
  color = 'darkGrey',
  inline = false,
  italic = false,
  bold = false,
  caption = false,
}) => {
  const cs = cx(
    styles.Text,
    color === 'darkGrey' && colors.darkGrey,
    color === 'white' && colors.white,
    color === 'green' && colors.green,
    color === 'darkGrey70' && colors.darkGrey70,
    color === 'darkGrey30' && colors.darkGrey30,
    color === 'darkGrey10' && colors.darkGrey10,
    color === 'white90' && colors.white90,
    color === 'white30' && colors.white30,
    color === 'green30' && colors.green30,
    color === 'green20' && colors.green20,
    color === 'yellow' && colors.yellow,
    color === 'red' && colors.red,
    align === 'center' && typography.alignCenter,
    align === 'justify' && typography.alignJustify,
    align === 'left' && typography.alignLeft,
    align === 'right' && typography.alignRight,
    italic && typography.fontStyleItalic,
    !italic && typography.fontStyleNormal,
    bold && typography.fontWeightBold,
    !bold && typography.fontWeightNormal,
  );

  console.log(cs)

  const Tag = inline ? 'span' : 'div';
  return (
    <Tag
      className={cs}
    >cs: {JSON.stringify(typography)}</Tag>
  )
};

Text.propTypes = {
  children: PropTypes.node.isRequired,
  align: PropTypes.oneOf(['left', 'right', 'center', 'justify']),
  color: PropTypes.oneOf([
    'darkGrey',
    'white',
    'green',
    'darkGrey70',
    'darkGrey30',
    'darkGrey10',
    'white90',
    'white30',
    'green30',
    'green20',
    'yellow',
    'red',
  ]),
  inline: PropTypes.bool,
  italic: PropTypes.bool,
  bold: PropTypes.bool,
  caption: PropTypes.bool,
};

export default Text;
