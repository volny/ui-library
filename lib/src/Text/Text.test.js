/**
 * @jest-environment node
 */

/* eslint-env jest */
import React from 'react'
import { create } from 'react-test-renderer'

import Text from './Text'

test('Text renders', () => {
  const tree = create(<Text>Text</Text>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Text inline creates a span instead of a p', () => {
  const tree = create(<Text inline>Text</Text>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Text italic creates italic text', () => {
  const tree = create(<Text italic>Text</Text>).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('font-style', 'italic')
})

test('Text bold creates bold text', () => {
  const tree = create(<Text bold>Text</Text>).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('font-weight', 'bold')
})

test('Text green adds green color', () => {
  const tree = create(<Text color="green">Text</Text>).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('color', '#080')
})
