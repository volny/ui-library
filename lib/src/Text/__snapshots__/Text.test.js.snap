// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text bold creates bold text 1`] = `
<div
  className="paragraph leftAlign darkGrey fontWeightBold fontStyleNormal"
>
  Text
</div>
`;

exports[`Text green adds green color 1`] = `
<div
  className="paragraph leftAlign green fontWeightNormal fontStyleNormal"
>
  Text
</div>
`;

exports[`Text inline creates a span instead of a p 1`] = `
<span
  className="paragraph leftAlign darkGrey fontWeightNormal fontStyleNormal"
>
  Text
</span>
`;

exports[`Text italic creates italic text 1`] = `
<div
  className="paragraph leftAlign darkGrey fontWeightNormal fontStyleItalic"
>
  Text
</div>
`;

exports[`Text renders 1`] = `
<div
  className="paragraph leftAlign darkGrey fontWeightNormal fontStyleNormal"
>
  Text
</div>
`;
